testng-core/src/main/java/org/testng/internal/invokers/ConfigInvoker.java

```diff
@@ -72,34 +75,49 @@ public ConfigInvoker(
    */
   public boolean hasConfigurationFailureFor(
       ITestNGMethod testNGMethod, String[] groups, IClass testClass, Object instance) {
+    return hasConfigurationFailureFor(null, testNGMethod, groups, testClass, instance);
+  }
+
+  @Override
+  public boolean hasConfigurationFailureFor(
+      ITestNGMethod configMethod,
+      ITestNGMethod testNGMethod,
+      String[] groups,
+      IClass testClass,
+      Object instance) {
     boolean result = false;
 
     Class<?> cls = testClass.getRealClass();
 
     if (m_suiteState.isFailed()) {
-      result = true;
-    } else {
-      boolean hasConfigurationFailures = classConfigurationFailed(cls, instance);
-      if (hasConfigurationFailures) {
-        if (!m_continueOnFailedConfiguration) {
-          result = true;
-        } else {
-          Set<Object> set = getInvocationResults(testClass);
-          result = set.contains(instance);
-        }
-        return result;
+      // if there were suite level failures, then short circuit here itself and report them.
+      return true;
+    }
+
+    boolean annotationFound = canDisregardConfigFailure(testClass);
+    boolean hasConfigurationFailures = classConfigurationFailed(cls, instance);
+    if (hasConfigurationFailures) {
+      if (annotationFound) {
+        // We were told to ignore failures via the annotation.
+        return false;
       }
-      // if method is BeforeClass, currentTestMethod will be null
-      if (m_continueOnFailedConfiguration && hasConfigFailure(testNGMethod)) {
-        Object key = TestNgMethodUtils.getMethodInvocationToken(testNGMethod, instance);
-        result = m_methodInvocationResults.get(testNGMethod).contains(key);
-      } else if (!m_continueOnFailedConfiguration) {
-        for (Class<?> clazz : m_classInvocationResults.keySet()) {
-          if (clazz.isAssignableFrom(cls)
-              && m_classInvocationResults.get(clazz).contains(instance)) {
-            result = true;
-            break;
-          }
+      if (m_continueOnFailedConfiguration) {
+        Set<Object> set = getInvocationResults(testClass);
+        result = set.contains(instance);
```

Comment:
U mean you want this to be added as an attribute to the annotations instead of having it as a separate annotation on its own ?

Reply:
What is the advantage of having `failurePolicy` as an attribute vs it being a completely standalone annotation on its own? The current annotations are configurable at runtime via annotation transformers and this behavior is NOT meant to be configurable via transformers because that kind of defeats the entire purpose (we have the suite attribute for that) and we would like this to be hard wired into the java class which can ONLY be reverted via a code change (similar to what `@Ignore` does today). Please let me know your thoughts on that @juherr